_posts/2021-01-11-strategies-for-small-focused-pull-requests.markdown

```diff
@@ -0,0 +1,119 @@
+
+We probably review infrastructural changes differently than we review implementation changes: 
+
+* Infrastructural work deserves scrutiny for the abstractions it introduces and how it might affect performance or future implementations. These kinds of changes introduce new patterns to the codebase and we want to make sure they're useful and usable patterns. 
+
+* An individual implementation gets more scrutiny on user-facing details. It's probably combining _existing_ patterns, so we'll spend less time looking at abstractions. We'll spend more time confirming it works for our users. 
+
+When a large PR combines wide, shallow, abstract work with tall, narrow, concrete work, it requires the reviewer to shift between two different mindsets. You might consider breaking your PR into two: one containing the wide infrastructural work, and one containing the tall implementation work. This allows reviewers to focus on abstractions in one PR and user-facing details in the other. 
+
+Some examples of infrastructural changes that could be separated from implementation work:
+
+* We introduced a seam to the code in order to make room for our implementation. 
+* We updated a dependency to take advantage of a new feature. 
+* We refactored before we started our implementation.
+
+## Separating an already-large PR
+
+It's natural for PRs to grow large. [Optimism bias][optimism-bias] affects our ability to estimate work often resulting in more code changes than we expected. A feature seems like it won't take much work until you get deeper and find complexity in places you hadn't considered. There's a lot of uncertainty when you start working on a feature and we'd need to model the entire problem to completion to know what the PR was going to look like before we started. A PR seems like it will be small until suddenly...it isn't anymore.
+
+This is what usually prevents developers from separating PRs — by the time you recognize the PRs could be de-tangled, it seems like a lot of effort to de-tangle them. 
+
+When you've got a PR/branch that contains multiple lines of work and you want to separate them, [`git rebase`][git-rebase] is your best friend. Rebasing enables you to rename, reorder, combine, and separate commits. Use `git rebase` to group your commits into one set for each branch you want to extract, and submit a PR for each smaller set of changes. 
+
+Good commit hygiene makes it easier to rebase commits. Commit small units of work so that they can be re-ordered and grouped, and apply clear messages to each commit in case you need to move it. While you might not _always_ separate/rebase PR branches, you'll appreciate small commits with clear messages when you do. 
+
+## conclusion
+
+somethiing something small PRs yay
```

Comment:
One thing I think you could reference here in the conclusion or elsewhere is that a lot of this chunking and dividing can happen during the ticketing/planning process. Well-scoped tickets = small and focused PRs. I think this idea could fit nicely at the end: Keeping this process in mind is critical through all steps of the development life cycle: from the planning process to implementation to monitoring after launch.